common/genericResourceURLParts.go

```diff
@@ -0,0 +1,116 @@
+package common
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/Azure/azure-storage-blob-go/azblob"
+	"github.com/Azure/azure-storage-file-go/azfile"
+
+	"github.com/Azure/azure-storage-azcopy/azbfs"
+)
+
+// GenericResourceURLParts is intended to be a generic solution to code duplication when using *URLParts
+// TODO: Consider returning to a "common" package structure for our SDKs
+//       (or, if we don't want to rely on a common package, create one that unifies URLParts and such into interfaces)
+//       The experience of trying to avoid code dupe is frankly quite poor and tends to result in more code dupe.
```

Comment:
common package is very problematic and will be avoided in the future. the files starting with 'zc' are the common files that are shared